src/utils/tree.ml

```diff
@@ -0,0 +1,51 @@
+type 'a tree = { node : 'a; children : 'a forest }
+and 'a forest = 'a tree list
+
+module type S = sig
+  type 'a t
+
+  val fold_left : f:('acc -> 'a -> 'acc) -> 'acc -> 'a t -> 'acc
+  val iter : f:('a -> unit) -> 'a t -> unit
+  val map : f:('a -> 'b) -> 'a t -> 'b t
```

Comment:
I'm using the `base` convention of labelling the function `f`, as I find it cleaner for this kind of functions, but if you want to keep the existing odoc conventions that's fine.

Reply:
I'm fine with either. We already have a mix. `f` is never really enlightening nor get punned, but it's sometimes nice to choose the order of the args (`map x ~f:(fun x -> very long function body)` or `x |> f1 |> f2 |> map ~f:apply`).